Tests/NIOTests/ByteBufferTest.swift

```diff
@@ -3052,3 +3052,45 @@ extension ByteBufferTest {
     }
     
 }
+
+// MARK: - ExpressibleByArrayLiteral init
+extension ByteBufferTest {
+    
+    func testCreateBufferFromArray() {
+        let bufferView: ByteBufferView = [0x00, 0x01, 0x02]
+        let buffer = ByteBuffer(ByteBufferView(bufferView))
+        
+        XCTAssert(buffer.readableBytesView == [0x00, 0x01, 0x02])
+    }
+
+    func testByteBufferViewEqualityWithRange() {
+        var buffer = self.allocator.buffer(capacity: 8)
+        buffer.writeString("AAAABBBB")
+        
+        let view = ByteBufferView(buffer: buffer, range: 2..<6) // 
+        let comparisonBuffer: ByteBufferView = [0x41, 0x41, 0x42, 0x42]
+
+        XCTAssert(view == comparisonBuffer)
+    }
+
+    func testInvalidBufferEqualityWithDifferentRange() {
+        var buffer = self.allocator.buffer(capacity: 4)
+        buffer.writeString("AAAA")
+        
+        let view = ByteBufferView(buffer: buffer, range: 0..<2)
+        let comparisonBuffer: ByteBufferView = [0x41, 0x41, 0x41, 0x41]
+
+        XCTAssertFalse(view == comparisonBuffer)
+    }
+
+    func testInvalidBufferEqualityWithDifferentContent() {
+        var buffer = self.allocator.buffer(capacity: 4)
+        buffer.writeString("AAAA")
+        
+        let view = ByteBufferView(buffer: buffer, range: 0..<4)
+        let comparisonBuffer: ByteBufferView = [0x41, 0x41, 0x00, 0x00]
+
+        XCTAssertFalse(view == comparisonBuffer)
```

Comment:
Some minor cleanups in your assertions: can you use `XCTAssertEqual` and `XCTAssertNotEqual`? We get substantially better test output from those.